docs/custom-filters.adoc

```diff
@@ -45,23 +149,163 @@ Instances are specific to the channel between a single client and a single broke
 
 It exists while the client remains connected.
 
+=== Filter Patterns
+
+Kroxylicious Protocol Filters support several patterns:
+
+1. <<Synchronously intercepting/mutating Requests and Responses>>
+2. <<Responding from a Request Filter without Proxying>>
+3. <<Sending new messages to Kafka that are not visible to the Client>>
```

Comment:
I assume this is a pointer to making additional requests such as the topic UUID -> TopicName mappings, so I wonder if something like `sending out of band messages to the target cluster`?